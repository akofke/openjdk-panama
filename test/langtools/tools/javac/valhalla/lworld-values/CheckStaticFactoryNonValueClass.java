/*
 * @test /nodynamiccopyright/
 * @summary Check that a value factory may not be declared for a non-value class
 * @compile/fail/ref=CheckStaticFactoryNonValueClass.out -XDrawDiagnostics -XDdev CheckStaticFactoryNonValueClass.java
 */

final class CheckStaticFactoryNonValueClass {
    final int x;

    __ValueFactory CheckStaticFactoryNonValueClass() { // error
        x = 10;
    }

    static __ValueFactory CheckStaticFactoryNonValueClass foo() { // error
       CheckStaticFactoryNonValueClass x = new CheckStaticFactoryNonValueClass();
       x.x = 100; // error.
       return x;
    }
}
