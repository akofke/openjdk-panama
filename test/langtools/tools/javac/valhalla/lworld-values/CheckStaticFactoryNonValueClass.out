CheckStaticFactoryNonValueClass.java:10:20: compiler.err.mod.not.allowed.here: staticvaluefactory
CheckStaticFactoryNonValueClass.java:14:59: compiler.err.value.factory.must.be.member.of.value.type
CheckStaticFactoryNonValueClass.java:16:9: compiler.err.cant.assign.val.to.final.var: x
3 errors
